scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOPipelineModel.scala

```diff
@@ -119,24 +177,29 @@ class H2OMOJOPipelineModel(override val uid: String)
       }
 
       builder.addRow(rowBuilder)
-      val output = mojoPipeline.transform(builder.toMojoFrame)
-      val predictions = output.getColumnNames.zipWithIndex.map {
-        case (_, i) =>
-          val columnData = output.getColumnData(i).asInstanceOf[Array[_]]
-          if (columnData.length != 1) {
-            throw new RuntimeException("Invalid state, we predict on each row by row, independently at this moment.")
-          }
-          columnData(0)
-      }
-      val content = if (getNamedMojoOutputColumns()) predictions else Array[Any](predictions)
-      new GenericRowWithSchema(content, schema)
+      val inputMojoFrame = builder.toMojoFrame
+
+      transformData(inputMojoFrame)
     }
-    swudf(function, schema)
+    swudf(function, schemaTransport)
   }
 
   override def copy(extra: ParamMap): H2OMOJOPipelineModel = defaultCopy(extra)
 
-  override def transform(dataset: Dataset[_]): DataFrame = applyPredictionUdf(dataset, modelUdf)
+  override def transform(dataset: Dataset[_]): DataFrame = {
+    val baseDf = applyPredictionUdf(dataset, modelUdf)
+
+    if (getWithContributions()) {
```

Comment:
The if-else piece of code can be minimized. only `.withColumn(getContributionCol(), col(s"${getTransportCol()}.${getContributionCol()}"))` is extra in the if branch